test/e2e/shared/aws.go

```diff
@@ -235,9 +240,11 @@ func (i *AWSInfrastructure) CreateInfrastructure() AWSInfrastructure {
 		i.CreateInternetGateway()
 	}
 	i.AllocateAddress()
-	i.CreateNatGateway("public")
-	if i.NatGateway != nil && i.NatGateway.NatGatewayId != nil {
-		WaitForNatGatewayState(i.Context, *i.NatGateway.NatGatewayId, 180, "available")
+	if i.ElasticIP != nil && i.ElasticIP.AllocationId != nil {
```

Comment:
Do we want to continue without creating a NAT GW?

Reply:
It would be caught here https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/suites/unmanaged/unmanaged_functional_test.go#L758 and here https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/suites/unmanaged/unmanaged_functional_test.go#L767

This should just eliminate those situations where AllocateAddress returns an allocation ID but the immediate lookup of the address based on that ID returns nothing.